pkg/pillar/cmd/zedagent/handlenetworkinstance.go

```diff
@@ -105,10 +106,14 @@ func prepareAndPublishNetworkInstanceInfoMsg(ctx *zedagentContext,
 		info.BridgeName = status.BridgeName
 		info.BridgeIPAddr = status.BridgeIPAddr
 
-		for mac, ip := range status.IPAssignments {
+		for mac, ips := range status.IPAssignments {
 			assignment := new(zinfo.ZmetIPAssignmentEntry)
 			assignment.MacAddress = mac
-			assignment.IpAddress = append(assignment.IpAddress, ip.String())
+			ipAddr := net.IP{}
+			if len(ips) != 0 {
+				ipAddr = ips[0]
```

Comment:
Why just the first one? Does some other struct need a list?

Reply:
fixed code to add all ip addresses to list.